status: 2023q2: mfsBSD building integration (#192)
freebsd/freebsd-doc: website/content/en/status/report-2023-04-2023-06/mfsbsd-building-integration.adoc

=== Integrate mfsBSD into the release building tools

Links: +
link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[Wiki Article] URL: link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[]

Contact: Soobin Rho <soobinrho@FreeBSD.org>

This project creates an additional target of the weekly snapshots of -current and -stable versions of mfsBSD images in the src/release makefile.

> The problem this project aims to address is that, since only the release versions of mfsBSD images are produced currently, they tend to get out of sync with the tools in base.

Reviewer: ```suggestion
Currently, only the release versions of mfsBSD images are produced.
This means they tend to get out of sync with the tools in base.
This project aims to address that problem.
```

Better flow by splitting it into 3 sentences.

Author: Very nice change! Thank you for the suggestion!

=== Integrate mfsBSD into the release building tools

Links: +
link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[Wiki Article] URL: link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[]

Contact: Soobin Rho <soobinrho@FreeBSD.org>

==== What is mfsBSD?

"link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[mfsBSD] is a toolset to create small-sized but full-featured mfsroot based distributions of FreeBSD that store all files in memory (MFS) [Memory File System] and load from hard drive, usb storage device or optical media.
It can be used for a variety of purposes, including diskless systems, recovery partitions and remotely overwriting other operating systems."

> mailto:mm@FreeBSD.org[Martin Matuska] is both the author of the white paper above and the maintainer of the link:https://github.com/mmatuska/mfsbsd[mfsBSD repository].

Reviewer 2: How about this?  ...is both the author of the link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[white paper] and the...

I know you already link to the white paper above, but it took me a moment to realize what you meant by saying the "white paper *above*".  Another option could be to say "is both the author of the white paper linked above...".

Reviewer 2: Other than this optional change it looks good to me.

Author: Absolutely! That makes the writing much nicer. Changed to `link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[white paper]` now. Thanks @Jehops!

=== Integrate mfsBSD into the release building tools

Links: +
link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[Wiki Article] URL: link:https://wiki.freebsd.org/SummerOfCode2023Projects/IntegrateMfsBSDIntoTheReleaseBuildingTools[]

Contact: Soobin Rho <soobinrho@FreeBSD.org>

==== Purpose of this project

This project creates an additional target of the weekly snapshots of -current and -stable versions of mfsBSD images in the src/release makefile.
Currently, only the release versions of mfsBSD images are produced.
This means they tend to get out of sync with the tools in base.
This project aims to address that problem.

> ==== What is mfsBSD?

Reviewer 3: I would move this section up: I think it makes more sense to first define mfsBSD, then talk about your mfsBSD project.

"mfsBSD is a toolset to create small-sized but full-featured mfsroot based distributions of FreeBSD that store all files in memory (MFS) [Memory File System] and load from hard drive, usb storage device or optical media.
It can be used for a variety of purposes, including diskless systems, recovery partitions and remotely overwriting other operating systems."

> (link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[])

Reviewer 3: We generally avoid putting links out of sentences (unless it is in a links list).

I suggest to transform it in `link:https://people.freebsd.org/~mm/mfsbsd/mfsbsd.pdf[mfsBSD]`, using the first occurence of the msfBSD term in the body of your report.

> Martin Matuska <mm@FreeBSD.org> is both the author of the white paper above and the maintainer of the mfsBSD repository: link:https://github.com/mmatuska/mfsbsd[]

Reviewer 3: Please use the `mailto` macro here:

`mailto:mm@FreeBSD[Martin Matuska]`

And use the link macro as following:

`link:https://github.com/mmatuska/mfsbsd[mfsBSD repository]`.

==== Location of this project

> This project is a GSoC 2023 (Google Summer of Code) project, under mentorship from Juraj Lutter <otis@FreeBSD.org> and Joseph Mingrone <jrm@FreeBSD.org>

Reviewer 3: Please use the `mailto` macro here as well.

As a GSoC project, the official coding period is between May 29, 2023 and August 28, 2023.

> As a humble beginner in the open-source community, the author of this project welcomes all comments / suggestions / PR's to the project repository, which is where all the code for this project will be located for the duration of the official GSoC coding period: link:https://github.com/soobinrho/freebsd-src/tree/integrate-mfsBSD-building[]

Reviewer 3: `link:https://github.com/soobinrho/freebsd-src/tree/integrate-mfs-building[project repository]`

Author: Awesome! I'll make corresponding changes when I get back home. Thank you for these suggestions; they are awesome changes for my status report!
